### PR TITLE
add owner to acl authorization

### PR DESCRIPTION
**Pull request by bourgeoa**
owner is defined as the subject in /.meta.
This is pre-existing in NSS

### CODE REVIEW

File: lib/handlers/allow.js [hunk 1]
```diff
@@ -68,6 +69,21 @@ function allow (mode) {
         }
       }
     }
+
+    // check user is owner. Find owner from /meta
+    let graph
+    try {
+      // do not check for permission. Owner has all rights.
```
Comment (csarven) [inline lib/handlers/allow.js:76]: Just  a heads-up that the owner has implicit Control privilege on all resources in the Storage, not necessarily Read-Write on the resource itself.

File: lib/handlers/allow.js [hunk 2]
```diff
@@ -68,6 +69,21 @@ function allow (mode) {
...
+      // TODO check permissions on .meta ???
+      graph = await ldp.getGraph(rootUrl + '/.meta')
+      const SOLID = $rdf.Namespace('http://www.w3.org/ns/solid/terms#')
+      const owner = await graph.any(null, SOLID('account'), $rdf.sym(rootUrl + '/'))
```
Comment (csarven) [inline lib/handlers/allow.js:80]: Not sure about the role that `solid:account` plays towards the role that `solid:owner` plays. `solid:account` typically refers to the root container / Storage

**Review by csarven — COMMENTED** [review]
Will the discovery of the solid:owner link relation from the root container be a separate PR?

File: lib/handlers/allow.js (hunk 2 above)
Comment (timbl) [reply lib/handlers/allow.js:80]: `solid:account` seems to be what NSS is already using for this.  So this fix can piggyb ack on that.

File: lib/handlers/allow.js (hunk 1 above)
Comment (timbl) [reply lib/handlers/allow.js:76]: Agree, only control, not necessarily read or write.

Comment (timbl) [reply lib/handlers/allow.js:76]: So something like
```
    if (mode === 'Control') {}
```
around this bit seems it would do that

**Review by timbl — CHANGES_REQUESTED** [review]

**Comment by bourgeoa** [comment]
> Agree, only control, not necessarily read or write.

This is not clear to me.
here is an ACL where users have only `Read` real example on solid-test running locahost as multiuser
https://alain.localhost:8443/public/text3.txt.acl
```
@prefix : <#>.
@prefix acl: <http://www.w3.org/ns/auth/acl#>.
@prefix foaf: <http://xmlns.com/foaf/0.1/>.
@prefix Gro: <ContactsTest/Group/Group.ttl#>.
@prefix tes: </testGroup.ttl#>.

:Read
    a acl:Authorization;
    acl:accessTo <text3.txt>;
    acl:agentClass foaf:Agent;
    acl:agentGroup Gro:this, tes:this;
    acl:mode acl:Read.```
```
If the `owner` wants to edit and change this ACL the owner in solidOs on NSS needs : 
- `Control` because it is an ACL
- `Read` to be able to GET the resource acl
- `Write` to PUT changes

 `Read` is only needed because of the display in solidOs, but `Write` is needed for PUT.

File: lib/handlers/allow.js (hunk 1 above)
Comment (bourgeoa) [reply lib/handlers/allow.js]: was resolved with a check that resource must be an acl.
`mode = Control` for acl's is done in `acl-checker.js`

**Comment by bourgeoa** [comment]
- when the webId is the owner Some test do return 200 ok if the acl is a bad acl (example with body = '')
- it used to be 403, and still is if the webId is not the owner).

Can this wait until implementation of WAC editor's draft ?
Test suite passes without any problem.

**Review by timbl — APPROVED** [review]
Looks good and I understand this is tested

**Review by michielbdejong — APPROVED** [review]
No objections from my side!

**Pull request merged**